core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -573,11 +575,23 @@ private void unparseOtherFunction(SqlWriter writer, SqlCall call, int leftPrec,
     case "STR_TO_DATE":
       unparseStrToDate(writer, call, leftPrec, rightPrec);
       break;
+    case "RPAD":
+    case "LPAD":
+      unparsePadding(writer, call, leftPrec, rightPrec);
+      break;
     default:
       super.unparseCall(writer, call, leftPrec, rightPrec);
     }
   }
 
+  private void unparsePadding(SqlWriter writer, SqlCall call, int leftPrec, int rightPrec) {
```

Comment:
can we move this method at some common location like PaddingFunctionUtil so that we can avoid repeated code?